Ensure frontend build process raises errors to sbt

jamespamplin opened the pull request:
Modifed frontend build task in sbt to properly throw errors when build task fails, ensures that builds fail on error.

Other enhancements:
- moved frontend build task to own sbt `AutoPlugin` for easier maintenance
- files can be ignored from build process (such as `hbs` as they're not used client-side yet), improves dev performance when modifying files
- extra build commands can be easily defined
- error output properly captured and will show in browser via Play

susiecoleman commented:
I don't know enough about sbt to check this fully but I'm happy to have a look through with someone else

markjamesbutler commented on project/FrontendBuildPlugin.scala:
```diff
@@ -0,0 +1,124 @@
+    val commands = (buildCommands in build in Assets).value
+
+    def runProcess(cmd: BuildCommand) = {
+      log.info(s"[${cmd.id}] Running `${cmd.command}`")
```
Logback parameters offer superior performance when executing on disabled log.  See http://logback.qos.ch/manual/architecture.html

markjamesbutler commented on project/FrontendBuildPlugin.scala:
```diff
@@ -0,0 +1,124 @@
+    val (products, errors) = syncIncremental(cacheDirectory, commands) { ops =>
+      val results = ops.map(op => op -> runProcess(op))
+
+      val opResults = results.map {
```
Add Type as discussed.

markjamesbutler commented:
:+1 with comment.

jamespamplin commented:
Removed anonymous types and functions for explicit classes and named functions, and added more comments on whats going on so that things are easier to understand.
